sklearn/tests/test_random_projection.py

```diff
@@ -80,19 +81,17 @@ def test_input_size_jl_min_dim():
 ###############################################################################
 # tests random matrix generation
 ###############################################################################
-def check_input_size_random_matrix(random_matrix):
-    assert_raises(ValueError, random_matrix, 0, 0)
-    assert_raises(ValueError, random_matrix, -1, 1)
-    assert_raises(ValueError, random_matrix, 1, -1)
-    assert_raises(ValueError, random_matrix, 1, 0)
-    assert_raises(ValueError, random_matrix, -1, 0)
+@pytest.mark.parametrize("random_matrix", all_random_matrix)
+@pytest.mark.parametrize("n_components, n_features",[(0,0),(-1,1),(1,-1),(1,0),(-1,0)])
+def test_input_size_random_matrix(random_matrix,n_components,n_features):
+    with pytest.raises(ValueError):
```

Comment:
Sorry, I read this incorrectly and thought it was a `test_*` function. I think using a for look like `check_input_with_sparse_random_matrix` would be the more minimal change:

```python
def check_input_size_random_matrix(random_matrix):
    inputs = [(0,0), (-1,1), (1,-1), (1,0), (-1,0)]
    for n_components, n_features in inputs:
        with pytest.raises(ValueError):
            random_matrix(n_components, n_features)
```

This way `test_basic_property_of_random_matrix` does not need to change.